lib/versioning/VersionID.js

```diff
@@ -152,4 +156,73 @@ function decode(str) {
     }
 }
 
-module.exports = { generateVersionId, getInfVid, encode, decode };
+/**
+ * Encode a versionId to obscure internal information contained
+ * in a version ID (less than 40 bytes).
+ *
+ * @param {string} str - the versionId to encode
+ * @return {string} - the encoded tinyVersionId
+ */
+function tinyEncode(str) {
+    const total = LENGTH_TS + LENGTH_SEQ;
+    const half = total / 2;
+    const part1 = Number(str.substring(0, half));
+    const part2 = Number(str.substring(half, total));
+    const part3 = Buffer.from(str.substring(total));
+    const enc1 = base62Integer.encode(part1);
+    const enc2 = base62Integer.encode(part2);
+    const enc3 = base62String.encode(part3);
+    return base62Integer.encode(enc1.length) +
+        enc1 +
+        base62Integer.encode(enc2.length) +
+        enc2 +
+        base62Integer.encode(enc3.length) +
```

Comment:
You may not need to include `enc3.length` since it is the last field, it must end at the end of the whole encoded string

Reply:
If I don't encode the length, I cannot detect if there is garbage at the end of the provided version ID, so the decoder will attempt to decode garbage (last error test case).